### ISSUE TITLE
multiprocessing error on macOS with Python3.8

### ISSUE THREAD

**Issue by cchall** [Libensemble/libensemble#502]
I recently updated to Python 3.8 on my Mac and found that libEnsemble examples no longer seemed to run when multiprocessing is being used. Trying examples/tutorials/tutorial_aposmm.py with `python tutorial_aposmm.py  --nworkers 3 --comms local` gives the following error:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/runpy.py", line 262, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/runpy.py", line 95, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/chall/research/github/libensemble/examples/tutorials/tutorial_aposmm.py", line 45, in <module>
    H, persis_info, flag = libE(sim_specs, gen_specs, exit_criteria, persis_info,
  File "/Users/chall/.pyenv/versions/venv380/lib/python3.8/site-packages/libensemble/libE.py", line 138, in libE
    return libE_funcs[comms_type](sim_specs, gen_specs, exit_criteria,
  File "/Users/chall/.pyenv/versions/venv380/lib/python3.8/site-packages/libensemble/libE.py", line 313, in libE_local
    wcomms = start_proc_team(nworkers, sim_specs, gen_specs, libE_specs)
  File "/Users/chall/.pyenv/versions/venv380/lib/python3.8/site-packages/libensemble/libE.py", line 285, in start_proc_team
    wcomm.run()
  File "/Users/chall/.pyenv/versions/venv380/lib/python3.8/site-packages/libensemble/comms/comms.py", line 266, in run
    self.process.start()
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/context.py", line 283, in _Popen
    return Popen(process_obj)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/chall/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable. 
```

After finding that a clean environment with Python 3.7.6 seemed to work without any issue I discovered that multiprocessing updated the default start method from fork to spawn on macOS in 3.8.
Setting at the top of the run script:
```
import multiprocessing
multiprocessing.set_start_method('fork') 
```
Does allow the example to run without error in 3.8. While forcing spawn on 3.7.6 causes the same error to now appear.

**Comment by jmlarson1** [comment]
@cchall Thanks for pointing out that multiprocessing's default start method is spawn instead of fork on macOS in 3.8.

We'll get this in a bugfix shortly.

**Comment by jlnav** [comment]
@cchall We're trying out a fix on our develop branch based on your feedback. If you're using a clone of the repository and the develop branch already, then `git pull` should incorporate the fix and other recent changes.

If you're using a release from PyPI or Conda, you can expect the fix in our upcoming 0.7.1 release.

Thank you again!

**Comment by cchall** [comment]
Thanks for the fix @jlnav!
